### DISCUSSION TITLE
Improve UI for input_datetime

### DISCUSSION THREAD

**Discussion by ghost** [home-assistant/frontend#11195] [Other feature requests]
*[move[bot]](https://github.com/apps/move) commented on Dec 1, 2018, 3:14 AM UTC:*

_[baflo](https://github.com/baflo) commented on Jan 4, 2018, 9:24 AM UTC:_

The UI for input_datetime - at least for the time input - is very basic. Especially on mobile devices the experience is not very well. It doesn't even handle jumping from the "hours" field to the "minutes" field when two digits were entered.

Using HTML5 input elements like `<input type="time"/>` would allow the app to use the OS's time input methods, which should be much more elegant.

Alternatively, I found this project: [https://www.webcomponents.org/element/fooloomanzoo/datetime-picker](https://www.webcomponents.org/element/fooloomanzoo/datetime-picker)

_This issue was moved by [iantrich](https://github.com/iantrich) from [home-assistant/home-assistant-polymer#785](https://github.com/home-assistant/home-assistant-polymer/issues/785)._

*This issue was moved by [iantrich](https://github.com/iantrich) from [home-assistant/ui-schema#209](https://github.com/home-assistant/ui-schema/issues/209).*

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Dec 1, 2018, 3:14 AM UTC:*

_[raccettura](https://github.com/raccettura) commented on Jan 10, 2018, 4:44 PM UTC:_

Might be a wider discussion but can there be perhaps a flag to decide when to polyfill vs use native OS input methods?

The reason I mention this is OS input methods don't work as well in all use cases, for example one that always annoys me is passcode for alarm in android pops up the full screen keyboard... hard to use with a wall mounted tablet. An option to reduce that to a polyfilled numeric input would be much more suited. For wall mount, I think in general the polymer based inputs are actually pretty nice with touch devices.

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Dec 1, 2018, 3:14 AM UTC:*

_[baflo](https://github.com/baflo) commented on Jan 24, 2018, 8:35 AM UTC:_

That flag you describe, would you add it as a configuration item? Or how was the flag be set?

**Comment by ghost** [comment]
*@iantrich commented on Sep 17, 2019, 7:58 PM UTC:*

[baflo](https://github.com/baflo) is this in reference to the entities card?

**Comment by ghost** [comment]
*@Misiu commented on Sep 27, 2019, 9:01 PM UTC:*

Maybe a picker similar to [this](https://material-ui-pickers.dev/demo/timepicker) would be a nice addition?

**Comment by Misiu** [comment]
There is another issue with input_datetime, as shown on the below image:
![ha_input2](https://user-images.githubusercontent.com/1741838/67757778-b6996a00-fa3c-11e9-94af-a5e1a46ad6c0.png)

on entities card, you get "normal" hour display - `08:00` but when You click on it a popup appears and it cuts off zeros from hours and minutes.

A proper picker would solve this problem.
